### PR TITLE
[New Term] C++ Math Functions: `fmod()`

### LINKED ISSUE #963: [Term Entry] C++ Math Functions: `fmod()`
### Reason/inspiration (optional)

Under `cpp/concepts/math-functions/terms/`, create a new term entry for the `fmod()` function,

- In the `terms/` folder, make a new folder called `fmod`
- Inside the `fmod` folder, create a new markdown file called `fmod.md`
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Let's include an "Example" section with a \```cpp code snippet.
- Lastly, let's include a "Codebyte Example" with a runnable \```codebyte/cpp

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by itsTodFod**
## Description

Added new term entry C++ Math Functions: fmod()

Closes #963

## Type of Change

Adding a new entry

## Checklist

- [x]  All writings are my own.
- [x]   My entry follows the Codecademy Docs style guide.
- [x]   My changes generate no new warnings.
- [x]   I have performed a self-review of my own writing and code.
- [x]   I have checked my entry and corrected any misspellings.
- [x]   I have made corresponding changes to the documentation if needed.
- [x]   I have confirmed my changes are not being pushed from my forked main branch.
- [x]   I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x]   Under "Development" on the right, I have linked any issues that are relevant to this PR.

### CODE REVIEW

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:37]: "~These~ The ~overloads~ overloaded functions ~effectively~ cast ~its~ the arguments to ~double~ a `double` type before ~calculations~ the calculation ~,~. ~except~ If ~at least~ one of the arguments is of type ~long double~ `long double` ~(in which case both are casted to long double instead)~, both arguments are cast to a `long double` type"

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:39]: Consider this statement instead:

"If the computation is successful, a floating-point remainder of the division  of `x` by `y` is returned as defined above."

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:40]: "~If the denominator y is zero~ the denominator is zero"

~NaN~ `NaN`

**Review by HishamT — CHANGES_REQUESTED** [review]
Hey, @itsTodFod, I have finished the first review of your entry.

**Comment by itsTodFod** [comment]
Made the changes as asked.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (HishamT) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:39]: `x` by `y`.

**Review by HishamT — APPROVED** [review]
Looks good! I just left some minor comments.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:41]:
```suggestion
> **Note:** The `<cmath>` header provides additional overloads for other combinations of arithmetic types (`double`, `float`, or `long double`). Overloaded functions cast the arguments to a `double` type before the calculation. If one of the arguments is a `long double` type, then both arguments are cast as `long double` types. 
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:53]:
```suggestion
int main() {
```

Most of our existing C++ entries seem to have the opening curly brace on the same line as the opening signature.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:62]:
```suggestion
  double x = 7.5, y = 2.1;
  double result = fmod(x, y);
  cout << "Remainder of " << x << "/" << y << " = " << result << endl;
    
  x = -17.50, y = 2.0;
  result = fmod(x, y);
  cout << "Remainder of " << x << "/" << y << " = " << result << endl;
    
  return 0;
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:92]:
```suggestion
int main() {
  double x = 7.5, y = 2.1;
  double result = fmod(x, y);
  cout << "Remainder of " << x << "/" << y << " = " << result << endl;
    
  x = -17.50, y = 2.0;
  result = fmod(x, y);
  cout << "Remainder of " << x << "/" << y << " = " << result << endl;
    
  return 0;
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:80]:
```suggestion
#include <iostream>
#include <cmath>
using namespace std;
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md:74]: I would go with something like this:

```suggestion
The following example is runnable and returns the floating-point remainder:
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:85]: Why don't we either `x` or `y` a different data type to demonstrate the casting nature of `<cmath>` and `fmod()`?

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:45]:
```suggestion
The following example uses `fmod()` to return the floating-point remainder of `x`/`y` as a `double` type:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@itsTodFod Finished with my review and left you some comments/suggestions. Tag me when you've finished implementing the feedback or if you have any questions. 😄

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md:85]: Sure, What about changing it to float?
Should i do it in Example, Codebyte Example or both?

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md:23]: Link underflow to an article of Codecademy or CPP Reference Sites?

Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md:23]: Use a collapsible?
<details>
  <summary>Click Click!</summary>
https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab
</details>

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md:85]: Committing changes with this issue yet to resolve. Will change as soon as i hear from you.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md]: Implemented.
Title box is not displayed as expected. Didn't make any significant changes to it though. Can you please look into the issue?
Thanks.

**Comment by itsTodFod** [comment]
@Dusch4593 Implemented.
Title box is not displayed as expected. Didn't make any significant changes to it though. Can you please look into the issue?
Thanks.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:3]:
```suggestion
Description: 'Returns the floating-point remainder of the division of two values, rounded towards zero.'
```

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md]: > Sure, What about changing it to float?
> Should i do it in Example, Codebyte Example or both?

Yeah, a float is fine and you can do this either in the Example, the Codebyte Example, or both. 😄

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md]: Hmm I would rather we have a link to a Codecademy resource about underflow. I'm not able to easily find one right now so what I would suggest is the following edit:

> If a range error occurs, then the correct result is rounded and returned.

Perhaps in a future PR, we can edit this entry with further info about underflow **_or_** we could create a new C++ concept about overflow/underflow. But for now, let's omit that part. The point in this sentence will remain in tact.

**Comment by Dusch4593** [comment]
Hey @itsTodFod!

> Title box is not displayed as expected. Didn't make any significant changes to it though. Can you please look into the issue?

It might have something to do with the missing quote mark at the end of the meta-`Description` (see my suggested change for this).

> How can i view the changes in Rich Mode on my local machine?
Currently, I need to push changes so that i can view those on Github.

This would depend on what software you are using to access the `docs` repo on your local machine (e.g., VS Code). One workaround I know of for previewing the markdown in Rich Mode is by copy/pasting into the textarea in https://markdownlivepreview.com/. You could use this to ensure what your writing will render as expected.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md]: Agreed.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (itsTodFod) [reply content/cpp/concepts/math-functions/terms/fmod/fmod.md]: Modified Codebyte Example so that now there is one of each type for learners. 😀

**Comment by itsTodFod** [comment]
Hi @Dusch4593 

> It might have something to do with the missing quote mark at the end of the meta-`Description` (see my [suggested change](https://github.com/Codecademy/docs/pull/1088#discussion_r993690716) for this).

I don't know how i missed that silly one but Thanks!

> This would depend on what software you are using to access the `docs` repo on your local machine (e.g., VS Code). One workaround I know of for previewing the markdown in Rich Mode is by copy/pasting into the textarea in https://markdownlivepreview.com/. You could use this to ensure what your writing will render as expected.

It doesn't display tables as tables but something is still better than nothing.
Thanks for the help. Just committing changes but still learnt a lot. Already feeling you are my Open Source Mentor. 😁

You can review the changes.

P.S. I would like to contribute more towards Codecademy in CPP and other ways i can under your guidance during and after the HacktoberFest.

**Comment by itsTodFod** [comment]
@Dusch4593 
Remainder : Modifications Completed. Waiting for approval.

File: content/cpp/concepts/math-functions/terms/fmod/fmod.md
Comment (Dusch4593) [inline content/cpp/concepts/math-functions/terms/fmod/fmod.md:39]:
```suggestion
int main() {
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
Hey @itsTodFod! Just a few last-minute comments/suggestions and then we'll be good to merge. 😄

**Comment by itsTodFod** [comment]
Done 😃

**Comment by Dusch4593** [comment]
Congratulations on making your first contribution to Docs, @itsTodFod! 🎉 🎊

The entry should now be live: https://www.codecademy.com/resources/docs/cpp/math-functions/fmod

Also, let me know if you'd like help with configuring your Codecademy profile so that it is credited on the entry 😄 

Screen Shot 2022-10-20 at 8 32 15 AM

**Pull request merged**
